libfreerdp/crypto/tls.c

```diff
@@ -159,6 +159,11 @@ static int bio_rdp_tls_read(BIO* bio, char* buf, int size)
 				break;
 
 			case SSL_ERROR_SYSCALL:
+				if (WSAGetLastError() == WSAEWOULDBLOCK)
```

Comment:
I think the last time we had problems in this area was caused by accessing the SSL BIOs from different threads iirc... maybe that leads you somewhere ...